code/game/jobs/job/support.dm

```diff
@@ -451,6 +451,16 @@
 	backpack_contents = list(
 		/obj/item/videocam/advanced = 1)
 
+/datum/outfit/job/librarian/post_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+	..()
+	if(!H.mind)
+		return
+	for(var/la in GLOB.all_languages)
+		var/datum/language/new_language = GLOB.all_languages[la]
+		if(new_language.flags & (HIVEMIND| NOLIBRARIAN))
+			continue
+		H.add_language(la)
```

Comment:
IMO, I think having a param on grant_all_babel_languages that also checks for the NOLIBRARIAN flag would be better. But this is fine if you prefer it

Reply:
ill take it for now, works the same way